### PR TITLE
[telemetry_autoscaling] Verify that the required custom resources are ready

### PR DESCRIPTION
**Pull request by myadla**
autoscaling test is giving unexpected response and needs  further Investigation

Made changes to add block rescue to gather all the information from autoscaling objects

### CODE REVIEW

**Comment by myadla** [comment]
Need to run this multiple times to make sure Autoscaling is enabled giving the correct output

**Comment by myadla** [comment]
recheck

**Comment by myadla** [comment]
recheck

File: ci/ansible.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [defaults]
-callbacks_enabled = custom_logger
+callbacks_enabled = custom_logger, junit
```
Comment (elfiesmelfie) [inline ci/ansible.cfg:2]: This is not required for this change
```suggestion
callbacks_enabled = custom_logger
```

File: ci/osp18_functional_tests.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@
   environment:
     KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
     PATH: "{{ cifmw_path }}"
+    JUNIT_TEST_CASE_PREFIX: "testsuites"
```
Comment (elfiesmelfie) [inline ci/osp18_functional_tests.yml:8]: This is also unrelated
```suggestion
```

File: ci/report_result.yml [hunk 3]
```diff
@@ -4,6 +4,7 @@
     cmd: |
       cp  {{ fvt_dir }}/test_run_result.out {{ logs_dir }}/test_run_result.log
       cp  {{ fvt_dir }}/summary_results.log {{ logs_dir }}/summary_results.log
+      cp  ~/.ansible.log/*.xml {{ logs_dir }}/junittest_run_result.xml
```
Comment (elfiesmelfie) [inline ci/report_result.yml:7]: This is unrelated to updating the autoscaling test
```suggestion
```

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 4]
```diff
@@ -17,14 +17,38 @@
...
+        cmd: |
+          oc get autoscaling autoscaling -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{"\n"}'
+      register: result
+      failed_when:
+        - result.stdout != "True"
+  rescue:
+    - name: view autoscaling objects
+      ansible.builtin.command:
+        cmd: |
+          oc get autoscaling -oyaml
```
Comment (elfiesmelfie) [inline roles/telemetry_autoscaling/tasks/test_services.yml:34]: Technically, you can run both commands here, on two lines.

```suggestion
          oc get autoscaling -oyaml
          oc get telemetry telemetry -oyaml
      ignore_errors: true
```
Then you only have to show one set of results.
I would recommend ignoring errors, for this so that the output is always shown, since errors might be useful as well.

**Review by elfiesmelfie — CHANGES_REQUESTED** [review]

**Comment by myadla** [comment]
recheck

File: ci/ansible.cfg (hunk 1 above)
Comment (myadla) [reply ci/ansible.cfg:2]: These commits are related to https://github.com/infrawatch/feature-verification-tests/pull/136

File: ci/report_result.yml (hunk 3 above)
Comment (myadla) [reply ci/report_result.yml:7]: https://github.com/infrawatch/feature-verification-tests/pull/136

File: ci/ansible.cfg (hunk 1 above)
Comment (elfiesmelfie) [reply ci/ansible.cfg:2]: They don't need to be included in this PR though. 
The only changes on this branch should be related to updating the autoscaling tests.

File: ci/report_result.yml (hunk 3 above)
Comment (elfiesmelfie) [reply ci/report_result.yml:7]: Since we have the [alternative junit plugin PR](https://github.com/infrawatch/feature-verification-tests/pull/132), the configuration of the junit plugin should be removed from this PR, to keep the scope limited to a single change (i.e. updating the autoscaling test)

**Review by elfiesmelfie — CHANGES_REQUESTED** [review]
Please remove the junit plugin configuration updates from this PR. Once that is done, we can merge this update to the autoscaling tests

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 5]
```diff
@@ -17,14 +17,28 @@
...
-# - name: Verify that custom resource AUTOSCALING is ready
-# It is returning an unexpected response and needs further debugging. will raise another PR to inspect this
-#   ansible.builtin.command:
-#     cmd: |
-#       oc get autoscaling autoscaling -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{"\n"}'
-#   register: result
-#   failed_when:
-#     - result.stdout != "True"
+#It is returning an unexpected response and needs further debugging
+- name: Enclose this task in a block/rescue to inspect further
```
Comment (elfiesmelfie) [inline roles/telemetry_autoscaling/tasks/test_services.yml:21]:
```suggestion
- name: Check that the autoscaling CR  is ready
```

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 6]
```diff
@@ -17,14 +17,28 @@
...
+          oc get autoscaling autoscaling -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{"\n"}'
+      register: result
+      failed_when:
+        - result.stdout != "True"
+  rescue:
+    - name: view autoscaling objects
+      ansible.builtin.command:
+        cmd: "{{ item }}"
+      register: result
+      with_items: [oc get autoscaling -oyaml, oc get telemetry telemetry -oyaml]
```
Comment (elfiesmelfie) [inline roles/telemetry_autoscaling/tasks/test_services.yml:35]:
```suggestion
      with_items: [oc get autoscaling autoscaling -oyaml, oc get telemetry telemetry -oyaml]
```

**Review by elfiesmelfie — APPROVED** [review]

**Comment by myadla** [comment]
Awaiting review to merge the change

**Review by lnatapov — APPROVED** [review]
lgtm

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 7]
```diff
@@ -17,14 +17,27 @@
...
+        cmd: |
+          oc get autoscaling autoscaling -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{"\n"}'
+      register: result
+      failed_when:
+        - result.stdout != "True"
+  rescue:
+    - name: view autoscaling objects
+      ansible.builtin.command:
+        cmd: "{{ item }}"
+      register: result
```
Comment (mgirgisf) [inline roles/telemetry_autoscaling/tasks/test_services.yml:33]: can you please use a different variable names  for different tasks to avoid conflicts?
for example: register: autoscaling_result

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 8]
```diff
@@ -17,14 +17,27 @@
...
 
-# - name: Verify that custom resource AUTOSCALING is ready
-# It is returning an unexpected response and needs further debugging. will raise another PR to inspect this
-#   ansible.builtin.command:
-#     cmd: |
-#       oc get autoscaling autoscaling -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}{"\n"}'
-#   register: result
-#   failed_when:
-#     - result.stdout != "True"
+- name: Check that the autoscaling CR  is ready
```
[the remark this answers is not in the document]
Comment (myadla) [reply roles/telemetry_autoscaling/tasks/test_services.yml]: Done

File: roles/telemetry_autoscaling/tasks/test_services.yml [hunk 9]
```diff
@@ -17,14 +17,27 @@
...
+      register: result
+      failed_when:
+        - result.stdout != "True"
+  rescue:
+    - name: view autoscaling objects
+      ansible.builtin.command:
+        cmd: "{{ item }}"
+      register: result
+      with_items: [oc get autoscaling autoscaling -oyaml, oc get telemetry telemetry -oyaml]
+      failed_when: result.rc >= 1
```
[the remark this answers is not in the document]
Comment (myadla) [reply roles/telemetry_autoscaling/tasks/test_services.yml]: There were some instances where the autoscaling result is giving unexpected response (Expected output is True)
but the job is completing successfully . Hence we added rescue block so that it gives the detailed output and also the job proceeds as expected without failing.

[the remark this answers is not in the document]
Comment (myadla) [reply roles/telemetry_autoscaling/tasks/test_services.yml]: Done

**Review by mgirgisf — APPROVED** [review]
Thanks Munesha.
